src/mpi/romio/adio/common/ad_close.c

```diff
@@ -61,7 +61,8 @@ void ADIO_Close(ADIO_File fd, int *error_code)
     }
 
     if (fd->hints) ADIOI_Free(fd->hints->ranklist);
-    if (fd->hints) ADIOI_Free(fd->hints->cb_config_list);
+    if (fd->hints)
```

Comment:
The change makes sense, I just think the nested if is unnecessary. C should guarantee left-to-right evaluation of the condition expression so that if we use `if (fd->hints && fd->hints->fs_hints.bg.bridgelist)` it will not chase the NULL pointer.